## One File for Testing Purpose

### GenomomFF.py

In [1]:
#Nrgc = GetNumber_rgc()  #WE GIVE THIS MANUALLY
#!wc -l  > Number_rgc

#Nsrl = GetNumber_srl() #WE  GIVE THIS MANUSALLY 
#!wc -l GC1806N2990_R1_V1_IonXpress_088.Fastq.sam.bam.sort.bam.rmdup.bam.sam.srl > Number_srl

#Loesstest_rgc() #R SCRIPT CODE WE WILL RUN WITH PYTHON

#Loesstest_srl() #RSCRIPT CODE 
#Arrangetest_rgc(Nrgc)
#Arrangetest_srl(Nsrl)

#Cal_enet_rgc()
#Cal_enet_srl()

#FINALLY CO RELATION
#CalCor()

In [2]:
#Total Line in rgc File, we use this Number in R code

In [3]:
!wc -l GC1806N2885_R1_V1_IonXpress_057.Fastq.sam.bam.sort.bam.rmdup.bam.sam_rgc.csv

10333 GC1806N2885_R1_V1_IonXpress_057.Fastq.sam.bam.sort.bam.rmdup.bam.sam_rgc.csv


In [4]:
#Total Line in srl File 

In [5]:
!wc -l GC1806N2885_R1_V1_IonXpress_057.Fastq.sam.bam.sort.bam.rmdup.bam.sam.srl

3881 GC1806N2885_R1_V1_IonXpress_057.Fastq.sam.bam.sort.bam.rmdup.bam.sam.srl


#### Lets Build a set of Function

In [6]:
import pandas as pd 
import subprocess 
import os
import sys
from scipy import stats

### Loesstest_rgc.py
This R Code accept `rgcbininfo***` file as input,<br>
From Bin info file it will abstract rgc bin file and through loop process (loop for all file in bininfo file) and give<br> Output File `ffyrgc***.csv` and `trgc***.csv` 2 File which is used later in ```      Code``` <br>
**ffyrgc900.csv   trgc900.csv**

In [7]:
def Loesstest_rgc():
    cmd = ['Rscript','loessgctest.R'] #Try to give wrong file 
    try:
        out = subprocess.Popen(cmd , stdout=subprocess.PIPE , stderr=subprocess.STDOUT)
        stdout, stderr = out.communicate()
        if stdout:
            sys.stderr.write(stdout)
            sys.exit(1)
    except:
        print("")      

In [8]:
Loesstest_rgc() #Must  give File Name Manually 
!ls ffy* trgc*
#!ls t*


ffyrgc900.csv  ffysrl900.csv  trgc900.csv


[1] "GC1806N2882"
[1] "GC1806N2883"
[1] "GC1806N2885"
[1] "GC1806N2889"
[1] "GC1806N2891"


In [9]:
import os
os.getcwd()

'/BiO/Alba/Enet/Enet_github/testing'

In [10]:
#!Rscript loessgctest.R

## loesstest_srl.py
###### Again we need to the same Process as above, `loesstest_srl.py` receive input file as **bininfo_list** 
##### and thorough sequencing(loop) and normalized process it produce  output file `ffysrl****.csv` and `tsr***.csv` file

In [11]:
def Loesstest_srl():
    cmd = ['Rscript', 'srlnormtest.R']
    try:
        out = subprocess.Popen(cmd , stdout=subprocess.PIPE , stderr=subprocess.STDOUT)
        stdout, stderr = out.communicate()
        if stdout:
            sys.stderr.write(stdout)
            sys.exit(1)
    except:
        print("")

In [12]:
Loesstest_srl()
!ls ffysrl* tsr*


ffysrl900.csv  tsrl900.csv


[1] "GC1806N2882"
[1] "GC1806N2883"
[1] "GC1806N2885"
[1] "GC1806N2889"
[1] "GC1806N2891"


In [13]:
#Run  R script Manually from Jupyter Notebook
#!Rscript srlnormtest.R

###### ##-------------------------------------------------------------------------------------------------------------------------------------------------##

## Arrangetest_rgc.py 
Receive `ffyrgc****.csv` file and `trgc8888.csv` file as input and give Output `temptestrgc` 

In [14]:
def Arrangetest_rgc():
    sp = {}
    ip = 'ffyrgc900.csv'
    input = open(ip,'r')
    check = 0
    for line in input:
        #pdb.set_trace()
        arr = str.split(line,',')
        if check > 0:
            key = check
            if key not in sp:
                sp[key] = []
            sp[key].append(arr[1][:-1])
        check += 1

    ip = 'trgc900.csv'
    input = open(ip,'r')
    check = 0
    for line in input:
        #pdb.set_trace()
        arr = str.split(line,',')
        if check > 0:
            key = check
            for k in range(1, len(arr)):
                if key not in sp:
                    sp[key] = []
                
                if k == len(arr)-1:
                    sp[key].append(arr[k][:-1])
                else:
                    sp[key].append(arr[k])
        check += 1
    out = 'temptestrgc'
    output = open(out,'w')
    for x in range(1, len(sp.keys())+1):
        output.write(str(sp[x][0])+',')
    #pdb.set_trace()
        for y in range(1, len(sp[x])):
        #pdb.set_trace()
            if y == len(sp[x])-1:
                output.write(sp[x][y]+'\n')
            else:
                output.write(sp[x][y]+',')
    output.close()
    
    #The above Codes Will Create temptestrgc file
    #temptestrgc file must be change in to y and x variable (temptestrgcy , temptestrgcrf)
    
    df = pd.read_csv("temptestrgc",header = None,) #since the file in same Folder
    
    (df.iloc[:,0]).to_csv("temptestrgcy",index = None ,header = False ) 
    #Write First Column to temptestrgcy file
    
    (df.iloc[:,1:]).to_csv("temptestrgcrf" ,index = None , header = False) #All column except 1
    #write a whole file from col 1 to end 

In [15]:
#Call the Arrangetest_rgc() function
Arrangetest_rgc()  #Output temptestrgc , temptestrgcy  ,temptestrgcrf
!ls tempt*
#We Created 3 File with this Code 

temptestrgc    temptestrgcy  temptestsrlrf  temptrainparargc.csv
temptestrgcrf  temptestsrl   temptestsrly   temptrainparasrl.csv


## arrangetest_srl()
arrangetest process for srl file, get 2 seprate file ffysrl****.csv and tsrl****.csv and produce one main and 2 sub files (dependent and non dependent variable)

In [16]:
import pandas as pd

In [17]:
def Arrangetest_srl():
    df_y = pd.read_csv("ffysrl900.csv",usecols=[1])
    df_y.to_csv("temptestsrly",index =False)

    df_x = pd.read_csv("tsrl900.csv")
    df_x = df_x.drop(columns=["Unnamed: 0"])
    df_x.to_csv("temptestsrlrf",index =False)

    merge = pd.concat([df_y,df_x], axis=1)
    merge.to_csv("temptestsrl")

In [18]:
#Call the Function  # temptestsrly  , temptestsrlrf  , temptestsrl , 

Arrangetest_srl()
#!rm temptestsrly temptestsrlrf temptestsrl
!ls temptest*

temptestrgc    temptestrgcy  temptestsrlrf
temptestrgcrf  temptestsrl   temptestsrly


###### ##-------------------------------------------------------------------------------------------------------------------------------------------------##

### cal_enet.py
Through cal_ent.py We are running `Rscript calenetrgc.R` where we put N = 10333 Manually in R source Code,
**calenetrgc.R** get input file **temptrainparargc.csv** as input and give Output file **enetrgc.csv**

In [19]:
#input y = temptestrgcy
#input y = temptestrgcrf
#out = enetrgc.csv

In [20]:
#!Rscript cal_enetrgc.R

In [21]:
def cal_enet_rgc():
    cmd = ['Rscript', 'cal_enetrgc.R']
    try:
        out = subprocess.Popen(cmd , stdout=subprocess.PIPE ,stderr=subprocess.STDOUT)
        stdout, stderr = out.communicate()
        if stdout:
            sys.stderr.write(stdout)
            sys.exit(1)
            
        print("srlnormtrain.R Ran Successfully")
    except:
        print("")

In [22]:
cal_enet_rgc() #Will Create file
! ls enetrg*


enetrgc.csv


Loading required package: Matrix
Loading required package: foreach
Loaded glmnet 2.0-8

[1] "#########1.bininfo_file check"
[1] "#########2.gmlnet parameter"
[1] "#########3.Matrix"


In [23]:
!Rscript cal_enetrgc.R

Loading required package: Matrix
Loading required package: foreach
Loaded glmnet 2.0-8

[1] "#########1.bininfo_file check"
[1] "#########2.gmlnet parameter"
[1] "#########3.Matrix"


## Cal_enet_srl.py()

In [24]:
#input y = temptestsrly
#rf =  temptestsrlrf
#Output = enetsrl.csv

In [25]:
def Cal_enet_srl():
    cmd = ['Rscript', 'cal_enetsrl.R']
    try:
        out = subprocess.Popen(cmd , stdout=subprocess.PIPE ,stderr=subprocess.STDOUT)
        stdout, stderr = out.communicate()
        print(stdout)
        if stdout:
            sys.stderr.write(stdout)
            sys.exit(1)
        print("srlnormtrain.R Ran Successfully")
    except:
        print("")
        #no need Except 

In [26]:
#Cal_enet_srl()
! ls enetsr*

ls: cannot access 'enetsr*': No such file or directory


In [27]:
!Rscript cal_enetsrl.R

Loading required package: Matrix
Loading required package: foreach
Loaded glmnet 2.0-8

[1] "#########1.bininfo_file check"
[1] "#########2.gmlnet parameter"
[1] "#########3.Matrix"
Error in x[i, ] %*% beta : 
  requires numeric/complex matrix/vector arguments
Execution halted


## CalCor .py


In [28]:
def CalCOR():
    sp1 = {}            #Blank Container
    ip = 'enetrgc.csv'  #
    input = open(ip,'r')
    check = 0
    for line in input:
        #pdb.set_trace()
        arr = str.split(line,',')
        if check > 0:
            key = check
            
            if key not in sp1:
                sp1[key] = []
            sp1[key].append(float(arr[2][:-1]))
            sp1[key].append(float(arr[1]))
        check += 1

    sp2 = {}
    ip = 'enetsrl.csv'
    input = open(ip,'r')
    check = 0
    for line in input:
        #pdb.set_trace()
        arr = str.split(line,',')
        
        if check > 0:
            key = check
            if key not in sp2:
                sp2[key] = []
            sp2[key].append(float(arr[2][:-1]))
            sp2[key].append(float(arr[1]))
        check += 1

    output = open("corr_out", 'a')
    
    output.write('Number'+'\t'+'Standard FF'+'\t'+'Predicted FF'+'\n')
    sf = []
    cf = []
    for x in range(len(sp1)):
        sf.append(sp1[x+1][0])
        cf.append((sp1[x+1][1]+sp2[x+1][1])/2)
        output.write(str(x+1)+'\t'+str(sp1[x+1][0])+'\t'+str((sp1[x+1][1]+sp2[x+1][1])/2)+'\n')

    cor = stats.pearsonr(sf, cf)
    output.write('#########################################'+'\n')
    output.write('Correlation:'+' '+str(cor[0])+'\n')
    output.close()

In [29]:
#CalCOR() #out Put File corr_out

### Arrangetest_rgc.py

In [30]:
def Arrangetest_rgc():
    sp = {}
    ip = 'ffyrgc900.csv'
    input = open(ip,'r')
    check = 0
    for line in input:
        #pdb.set_trace()
        arr = str.split(line,',')
        if check > 0:
            key = check
            if key not in sp:
                sp[key] = []
            sp[key].append(arr[1][:-1])
        check += 1

    ip = 'trgc900.csv'
    input = open(ip,'r')
    check = 0
    for line in input:
        #pdb.set_trace()
        arr = str.split(line,',')
        if check > 0:
            key = check
            for k in range(1, len(arr)):
                if not sp.has_key(key):
                    sp[key] = []
                
                if k == len(arr)-1:
                    sp[key].append(arr[k][:-1])
                else:
                    sp[key].append(arr[k])
        check += 1


    out = 'temptestrgc'
    output = open(out,'w')
    for x in range(1, len(sp.keys())+1):
        output.write(str(sp[x][0])+',')
    #pdb.set_trace()
        for y in range(1, len(sp[x])):
        #pdb.set_trace()
            if y == len(sp[x])-1:
                output.write(sp[x][y]+'\n')
            else:
                output.write(sp[x][y]+',')
    output.close()
    
    #The above Codes Will Create temptestrgc file
    #temptestrgc file must be change in to y and x variable (temptestrgcy , temptestrgcrf)
    
    df = pd.read_csv("temptestrgc",header = None,) #since the file in same Folder
    
    (df.iloc[:,0]).to_csv("temptestrgcy",index = None ,header = False ) 
    #Write First Column to temptestrgcy file
    
    (df.iloc[:,1:]).to_csv("temptrainrf" ,index = None , header = False) #All column except 1
    #write a whole file from col 1 to end 

### Arrangetest_srl()

In [31]:
#It is better to keep arrangetest_srl.py in another fiel and import as Module
def Arrangetest_srl():
    sp = {}
    ip = 'ffysrl900.csv'
    input = open(ip,'r')
    check = 0
    for line in input:
        arr = str.split(line,',')
        if check > 0:
            key = check
            if key not in sp:
                sp[key] = []
            sp[key].append(arr[1][:-1])
        check += 1

    ip = 'tsrl900.csv'
    input = open(ip,'r')
    check = 0
    for line in input:
        arr = str.split(line,',')
        if check > 0:
            key = check
            for k in range(1, len(arr)):
                
                if key not in sp:
                    sp[key] = []
                if k == len(arr)-1:
                    sp[key].append(arr[k][:-1])
                else:
                    sp[key].append(arr[k])
        check += 1

    out = 'temptestsrl'
    output = open(out,'w')
    for x in range(1, len(sp.keys())+1):
        output.write(str(sp[x][0])+',')
        for y in range(1, len(sp[x])):
            if y == len(sp[x])-1:
                output.write(sp[x][y]+'\n')
        else:
            output.write(sp[x][y]+',')
    output.close()
    
    #######################################################################################################
    ################# Since temptestsrl file Created lets do x and Y variable#############################
    
    df = pd.read_csv("temptestsrl",header = None,) #since the file in same Folder
    
    (df.iloc[:,0]).to_csv("temptestsrly",index = None ,header = False ) 
    #Write First Column to temptestrgcy file
    
    (df.iloc[:,1:]).to_csv("temptestsrlf" ,index = None , header = False) #All column except 1
    #write a whole file from col 1 to end 

In [32]:
# Lests Check srlist File 
#! ls -alh *.srl > srlist  #This can also be update by Python os mudule

In [33]:
import os
def Check(last_6_character="sam.srl"): #sam.srl or rgc.csv
    y = os.listdir()
    ln =0
    for i in y:
        #print(i[-7:])
        if i[-7:]==last_6_character:
            ln +=1
    return ln

In [34]:
#Check("rgc.csv") #default is sam.srl
Check()

5

## Define Remove ()